### PR TITLE
docs: Document kafka sink cdc output

### LINKED ISSUE #3541: Document Sink CDC format
Document Materialize's CDC payload format

### PR DESCRIPTION
**Pull request by ruchirK**
Closes #3541

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3673)

### CODE REVIEW

File: doc/user/content/overview/api-components.md [hunk 1]
```diff
@@ -154,11 +155,23 @@ quickly, e.g. `JOIN`s.
...
+## Sinks
+
+Sinks are the inverse of sources and represent a connection to an external stream
+where Materialize outputs data. When a user defines a sink over a materialized view
+or source, Materialize automatically generates the required schema and writes down
+the stream of changes to that view or source. In effect, Materialize sinks act as
+change data capture (CDC) producers for the given source or view.
+
+Currently, Materialize only supports sending sink data to Kafka or Avro OCFs,
+encoded in Avro with the Debezium diff envelope.
```
[the remark this answers is not in the document]
Comment (ruchirK) [reply doc/user/content/overview/api-components.md:167]: added a description inline

[the remark this answers is not in the document]
Comment (sploiselle) [reply doc/user/content/overview/api-components.md:167]: Awesome! Would you mind throwing a link to the section you added?

File: doc/user/content/sql/create-sink.md [hunk 2]
```diff
@@ -63,7 +64,60 @@ Any further updates to these results are produced at the time when they occur. T
...
+        ]
+        },
+        { "name": "after", "type": ["row", "null"] }
+    ]
+}
+```
+
+Note that:
+
+- You don't need to manually create this schema. Materialize will generate it for you.
```
Comment (sploiselle) [inline doc/user/content/sql/create-sink.md:109]: nit: s/will generate/generates/

**Comment by ruchirK** [comment]
> Ah, this is great! Some minor copy editing nits, but this is good to go 🤓

nit away by all means

**Pull request merged**